File: app/src/main/java/com/example/java_viewmodel/contents/item/ItemDao.java
package com.example.java_viewmodel.contents.item;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ItemDao {
    @Query("SELECT * FROM item_t")
    LiveData<List<ItemEntity>> getAllItems();

    @Query("SELECT * FROM item_t WHERE init_count >= :minInitCount  ORDER BY number ASC")
    LiveData<List<ItemEntity>> getParticularItems(long minInitCount);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(ItemEntity entity);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertList(List<ItemEntity> entityList);

    @Query("DELETE FROM item_t")
    void deleteAll();
}

File: app/src/main/java/com/example/kt_viewmodel/contents/item/ItemDao.kt
package com.example.kt_viewmodel.contents.item

import androidx.room.Dao
import androidx.lifecycle.LiveData
import androidx.room.Insert
import com.example.kt_viewmodel.contents.item.ItemEntity
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface ItemDao {
    @Query("SELECT * FROM item_t")
    fun getAllItems(): LiveData<List<ItemEntity>>

    @Query("SELECT * FROM item_t WHERE init_count >= :minInitCount  ORDER BY number ASC")
    fun getParticularItems(minInitCount: Long): LiveData<List<ItemEntity>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(entity: ItemEntity)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertList(entityList: List<ItemEntity>)

    @Query("DELETE FROM item_t")
    suspend fun deleteAll()
}